### ISSUE TITLE
Usage mandatory metric

### ISSUE THREAD

**Issue by bhack** [ildoonet/pytorch-gradual-warmup-lr#3]
Is in the usage example required a mandatory metric param?
https://github.com/ildoonet/pytorch-gradual-warmup-lr#usage

**Comment by ildoonet** [comment]
@bhack I don't understand exactly. Could you elaborate more?

**Comment by bhack** [comment]
In the readme scheduler_warmup.step() is without parameters but I think that at https://github.com/ildoonet/pytorch-gradual-warmup-lr/blob/master/warmup_scheduler/scheduler.py#L54 metric cannot be none.

**Comment by ildoonet** [comment]
The parameter 'epoch' can be none.

Also I fix some bugs when epoch parameter is provided.

**Comment by bhack** [comment]
We still need to send Metric param right? https://github.com/ildoonet/pytorch-gradual-warmup-lr/blob/master/warmup_scheduler/scheduler.py#L60

**Comment by ildoonet** [comment]
YES, If you use ReduceLROnPlateau(https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.ReduceLROnPlateau).

But No, if you use other schedulers(eg. cosine annealing) after gradual warmup.

I'll update readme accordingly.
